File: CleanJava/app/src/integrationTests/java/com/vascome/fogtail/FogtailIntegrationRobolectricTestRunner.java
package com.vascome.fogtail;

import android.support.annotation.NonNull;

import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

// Custom runner allows us set config in one place instead of setting it in each test class.
public class FogtailIntegrationRobolectricTestRunner extends RobolectricTestRunner {

    private static final int SDK_EMULATE_LEVEL = 26;

    public FogtailIntegrationRobolectricTestRunner(@NonNull Class<?> classRun) throws Exception {
        super(classRun);
    }
    @NonNull
    public static FogtailApplication fogtailApplication() {
        return (FogtailApplication) RuntimeEnvironment.application;
    }
}

File: CleanKotlin/app/src/integrationTests/java/com/vascome/fogtail/FogtailIntegrationRobolectricTestRunner.kt
package com.vascome.fogtail

import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment

// Custom runner allows us set config in one place instead of setting it in each test class.
class FogtailIntegrationRobolectricTestRunner @Throws(Exception::class)
constructor(classRun: Class<*>) : RobolectricTestRunner(classRun) {
    companion object {

        private val SDK_EMULATE_LEVEL = 26
        fun fogtailApplication(): FogtailApplication {
            return RuntimeEnvironment.application as FogtailApplication
        }
    }
}
